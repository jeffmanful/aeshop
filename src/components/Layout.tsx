import React from 'react';
import styled from 'styled-components';
import Header from './Header';

const ContentBox = styled.div`
  padding: 1rem;
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-wrap: wrap;
  max-width: 1200px;
`;
ContentBox.displayName = 'ContentBox';

const Layout: React.FC = ({ children }): JSX.Element => {
  return (
    <>
      <Header />
      <ContentBox>
        {children}
      </ContentBox>
    </>
  )
}

export default Layout;
