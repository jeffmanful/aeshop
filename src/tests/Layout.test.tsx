import React from 'react'
import Layout from '../components/Layout'
import { render } from '@testing-library/react'

test('It renders', () => {
  const { asFragment, queryByText } = render(
    <Layout>Body</Layout>
  )

  expect(asFragment()).toMatchSnapshot()
  expect(queryByText('Body')).toBeInTheDocument()
  expect(queryByText('Aeshop')).toBeInTheDocument()
})
