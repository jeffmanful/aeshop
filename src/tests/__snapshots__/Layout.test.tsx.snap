// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders 1`] = `
<DocumentFragment>
  <header
    class="sc-AxjAm ehXquI"
  >
    <h1>
      Aeshop
    </h1>
    <div
      class="sc-AxirZ dTDYBw"
    >
      <div
        style="display: flex;"
      >
        <img
          alt="liked"
          src="liked.svg"
          style="margin: 5px;"
        />
        <p>
          0
        </p>
      </div>
      <div
        class="sc-AxiKw juaMsy"
      />
    </div>
  </header>
  <div
    class="sc-AxhUy xbsSu"
  >
    Body
  </div>
</DocumentFragment>
`;
